File: UFOAppAngular/Models/Observasjon.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace UFOAppAngular.Models
{
    public class Observasjon
    {
        public int Id { get; set; }
        [RegularExpression(@"[a-zA-ZæøåÆØÅ. \-]{2,30}")]
        public string KallenavnUFO { get; set; }
        [RegularExpression(@"[0-9a-zA-ZæøåÆØÅ:. \-]{2,250}")]
        public DateTime TidspunktObservert { get; set; }
        [RegularExpression(@"[a-zA-ZæøåÆØÅ. \-]{2,30}")]
        public string KommuneObservert { get; set; }
        [RegularExpression(@"[a-zA-ZæøåÆØÅ. \-]{2,30}")]
        public string BeskrivelseAvObservasjon { get; set; }
        [RegularExpression(@"[0-9a-zA-ZæøåÆØÅ. \-]{2,250}")]
        public string Modell { get; set; }
        [RegularExpression(@"[a-zA-ZæøåÆØÅ. \-]{2,30}")]
        public string FornavnObservator { get; set; }
        [RegularExpression(@"[a-zA-ZæøåÆØÅ. \-]{2,30}")]
        public string EtternavnObservator { get; set; }
        [RegularExpression(@"[0-9]{8}")]
        public string TelefonObservator { get; set; }
        [RegularExpression(@"[0-9a-zA-ZøæåØÆÅ. \-.@ ]{2,30}")]
        public string EpostObservator { get; set; }

        public static implicit operator Observasjon(Observasjon v)
        {
            throw new NotImplementedException();
        }
    }
}

File: UFOAppAngular/ClientApp/src/app/Observasjon.ts
export class Observasjon {
    id: number;
    kallenavnUFO: string;
    tidspunktObservert: Date;
    kommuneObservert: string;
    beskrivelseAvObservasjon: string;
    modell: string;
    fornavnObservator: string;
    etternavnObservator: string;
    telefonObservator: string;
    epostObservator: string;
}

export class UFO {
    id: number;
    kallenavn: string;
    modell: string;
    gangerObservert: string;
    sistObservert: Date;
}

export class Observator {
    id: number;
    fornavn: string;
    etternavn: string;
    telefon: string;
    epost: string;
    antallRegistrerteObservasjoner: number;
    sisteObservasjon: Date;
}

export class Bruker {
    brukernavn: string;
    passord: string;
}

